.github/workflows/docker.yaml

```diff
@@ -0,0 +1,112 @@
+
+# Set to 'yes' to open a tunnel to GitHub's VMs through tmate on failures.
+# Also increase timeout-minutes for the relevant OS when debugging remotely.
+env:
+  CHEVAH_DEBUG: 'no'
+
+# Using a job name that doesn't contain the OS name, to minimize the risk of
+# confusion with the OS names of the containers, which are the relevant ones.
+jobs:
+  latest:
+    runs-on: ubuntu-latest
+    container: ${{ matrix.container }}
+    strategy:
+      fail-fast: false
+      matrix:
+        container: [ 'alpine:3.12', 'centos:8.2.2004', 'centos:5.11' ]
+    timeout-minutes: 30
+    steps:
+
+    # To avoid duplicating the other steps, setup steps are allowed to fail.
+    # First 'run' command should only run successfully on the intended target.
+    # OpenSSL gets updated by apk, but that is the Alpine way, so it's fine.
+    - name: Alpine 3.12 setup
+      continue-on-error: true
+      run: |
+        grep ^3\.12 /etc/alpine-release
+        apk update
+        apk upgrade
+        apk add git curl bash openssh-client
+        curl -o /usr/local/bin/paxctl https://binary.chevah.com/third-party-stuff/alpine/paxctl-3.12
+        chmod +x /usr/local/bin/paxctl
+
+    # To avoid duplicating the other steps, setup steps are allowed to fail.
+    # First 'run' command should only run successfully on the intended target.
+    # Stick to CentOS 8.2 as OpenSSL got updated in 8.3 from 1.1.1c to 1.1.1g.
+    - name: CentOS 8.2 setup
+      continue-on-error: true
+      run: |
+        grep "release 8.2.2004 (Core)" /etc/redhat-release
+        sed -i s/^mirrorlist=/#mirrorlist=/ /etc/yum.repos.d/*.repo
+        sed -i s@^#baseurl=http://mirror.centos.org/\$contentdir/\$releasever/@baseurl=https://vault.centos.org/8.2.2004/@ /etc/yum.repos.d/*.repo
+        yum -y upgrade
+        yum -y install git curl openssh-clients
+
+    # To avoid duplicating the other steps, setup steps are allowed to fail.
+    # First 'run' command should only run successfully on the intended target.
+    - name: CentOS 5.11 setup
+      continue-on-error: true
+      run: |
+        grep "release 5.11 (Final)" /etc/redhat-release
+        sed -i s/^mirrorlist=/#mirrorlist=/ /etc/yum.repos.d/*.repo
+        sed -i s@^#baseurl=http://mirror.centos.org/centos/\$releasever/@baseurl=http://vault.centos.org/5.11/@ /etc/yum.repos.d/*.repo
+        yum -y upgrade
+        # Use http://binary.chevah.com/third-party-stuff/centos5/tuxad/
+        # when tuxad.de dissapears, it has the minimum required stuff.
+        rpm -i http://www.tuxad.de/rpms/tuxad-release-5-1.noarch.rpm
+        yum -y install curl openssh-clients gcc44 make m4 patch unzip wget
+        ln -s /usr/bin/gcc44 /usr/local/bin/gcc
+        wget --mirror --no-parent https://binary.chevah.com/third-party-stuff/centos5/endpoint/
+        cd binary.chevah.com/third-party-stuff/centos5/endpoint/
+        rpm -i local-perl-*.rpm
+        rpm -i --nodeps git{-core,}-2.5.0-1.ep.x86_64.rpm
+
+    - name: Clone repo independently
```

Comment:
Why not use `- uses: actions/checkout@v2` ? Just asking ? :)

Reply:
Because `actions/checkout@v2` doesn't work on CentOS 5, for example.

Reply:
ah... right.... as it needs nodejs and centos might not have the right  version of nodejs.

just a suggestion...maybe see if nodejs can be updated inside centos and then I expect that github actions will run ok.

but this can be merged as is now.

Reply:
Don't think so, the glibc version required by nodejs is way newer than the one on CentOS 5. 

To give an idea of what kind of a task is to have recent software built on that platform, imagine that this CentOS 5 was built on 2007 when the kernel was 2.6, glibc was 2.5, gcc 4.4 is only available as a backport, etc.